### ISSUE TITLE
Is there away to split build_references into sections

### ISSUE THREAD

**Issue by DyfanJones** [r-lib/pkgdown#2293]
Hi All,

This is a weird question, is there away to split `build_reference` into sections? 

Background information:

I would like to automate the build of [paws]( https://github.com/paws-r/paws) website. However currently the build is taking over 6 hours which causes the github actions job to timeout (https://github.com/DyfanJones/dev.paws-r.github.io/actions/runs/4558343201)

It would be great if the website build could be split over multiple github actions jobs.

Website Build code: https://github.com/DyfanJones/dev.paws-r.github.io/blob/5f47070/docs.R

Any help/advice with this would be amazing.

**Comment by maelle** [comment]
Interesting question :dog: :feet: Some thoughts
- you can split the reference page into sections but only for easier navigation, not for building in different parts https://pkgdown.r-lib.org/reference/build_reference.html#reference-index
- do you know which examples take the most time? Could they be wrapped in dontrun, and instead they would be shown in a pre-built vignette or article? (see https://ropensci.org/blog/2019/12/08/precompute-vignettes/ and if you use https://usethis.r-lib.org/reference/use_release_issue.html#ref-usage you can create an `release_bullets()` function with a reminder to update the prebuilt docs)

**Comment by maelle** [comment]
Related (and closed + not implemented) https://github.com/r-lib/pkgdown/issues/805

**Comment by maelle** [comment]
Related to your docs but not pkgdown, the whole manual is rendered on R-universe https://paws-r.r-universe.dev/paws#

See https://ropensci.org/blog/2023/02/27/runiverse-discovering/#all-the-documentation

**Comment by DyfanJones** [comment]
Hi @maelle sorry about the slow reply.  

- Do you know which examples take the most time? I don't think it is a specific section that is being slow just a lot of stuff to build.
- Related to your docs but not pkgdown, the whole manual is rendered on R-universe https://paws-r.r-universe.dev/paws# I am happy r-universe offers this functionality however with paws the documentation is split into 2 versions.  Shortened version (to reduce package size for cran) and the Long version. Sadly I believe r-universe will be using the shortened version.

One a positive note I believe I have figured out how to build the site in sections. https://github.com/DyfanJones/dev.paws-r.github.io/actions/runs/4616292943. 

In short:

1.  Build Initial index to get css settings etc..
2. Split `build_reference`  by giving each job a selection of topics to build (I used a stripped down version of `build_reference` however I believe `build_reference` will work just fine).
3. Combine all topics together
4. Rebuild index (this is optional and only really needed for paws)

**Comment by maelle** [comment]
Ah, nice that you found a workaround! :rocket: Interesting that there are a short and a long version of your docs. :eyes: 
Should we close this?

**Comment by DyfanJones** [comment]
Closing ticket, thanks for all your help with this
